### PR TITLE
Fixes related to #1816

### PR DESCRIPTION
**Pull request by NathanJPhillips**
Since #1816, when an implementation of `java.lang.Object` is provided - for example when the models library is included in Deeptest then JBMC will attempt to load it twice.
This fix also makes JBMC slightly more resilient to passing two command line parameters that create instances of `java_languaget` that load the same class. This is not a supported configuration but is also used sometimes in Deeptest.
Finally JBMC previously ignored invalid jar files passed in the class path, this PR restores that behaviour.

### CODE REVIEW

**Comment by thk123** [comment]
Bump: diffblue/test-gen#1636

File: regression/cbmc-java/invalid_classpath/test-path.desc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CORE symex-driven-lazy-loading-expected-failure
+Test.class
+--classpath ./NotHere
+^EXIT=6$
+^SIGNAL=0$
+^the program has no entry point$
+^failed to load class `Test'$
+--
+--
+symex-driven lazy loading should emit "the program has no entry point" but currently doesn't
```
Comment (thk123) [inline regression/cbmc-java/invalid_classpath/test-path.desc:10]: Probably worth raising a ticket for this

Comment (NathanJPhillips) [reply regression/cbmc-java/invalid_classpath/test-path.desc:10]: @smowton - have you got this covered?

**Review by thk123 — COMMENTED** [review]
Changes look fine, will approve once TG passes

File: regression/cbmc-java/invalid_classpath/test-path.desc (hunk 1 above)
Comment (smowton) [reply regression/cbmc-java/invalid_classpath/test-path.desc:10]: done

File: regression/cbmc-java/invalid_classpath/test-jar.desc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+CORE symex-driven-lazy-loading-expected-failure
+Test.class
+--classpath ./NotHere.jar
+^EXIT=6$
+^SIGNAL=0$
+^the program has no entry point$
+^failed to load class `Test'$
+--
```
Comment (smowton) [inline regression/cbmc-java/invalid_classpath/test-jar.desc:8]: This is a bit cryptic, so worth explaining what you're testing for and why this is the expected behaviour (perhaps in a followup PR considering the urgency)

**Review by smowton — APPROVED** [review]

**Review by thk123 — APPROVED** [review]
Only Mac builds not yet passed due to travis backlog

**Pull request merged**